src/librustc_driver/lib.rs

```diff
@@ -324,34 +325,53 @@ pub trait CompilerCalls<'a> {
 #[derive(Copy, Clone)]
 pub struct RustcDefaultCalls;
 
+fn handle_explain(code: &str,
+                  descriptions: &diagnostics::registry::Registry,
+                  output: ErrorOutputType) {
+    let normalised = if !code.starts_with("E") {
+        format!("E{0:0>4}", code)
+    } else {
+        code.to_string()
+    };
+    match descriptions.find_description(&normalised) {
+        Some(ref description) => {
+            // Slice off the leading newline and print.
+            print!("{}", &description[1..]);
+        }
+        None => {
+            early_error(output, &format!("no extended information for {}", code));
+        }
+    }
+}
+
+fn check_cfg(sopts: &config::Options,
+             output: ErrorOutputType) {
+    fn is_meta_list(item: &ast::MetaItem) -> bool {
+        match item.node {
+            ast::MetaItem_::MetaList(..) => true,
+            _ => false,
+        }
+    }
+
+    if sopts.cfg.iter().any(|item| is_meta_list(&*item)) {
+        early_error(output, "predicates are not allowed in --cfg");
```

Comment:
Sounds good. I'll make the change after finding out how to print more than one error, since the user could have passed multiple offending flags. (This is my first time in the compiler's code.)